internal/commands/generate.go

```diff
@@ -5,50 +5,62 @@ import (
 	"io/ioutil"
 	"os"
 	"path"
-	"strings"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 
 	"github.com/plexsystems/promdoc/internal/rendering"
 )
 
 // NewGenerateCommand creates a new generate command
 func NewGenerateCommand() *cobra.Command {
 	cmd := cobra.Command{
-		Use:   "generate <output-dir>",
+		Use:   "generate",
 		Short: "Generate documentation from a given folder",
-		Args:  cobra.ExactArgs(1),
+
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := viper.BindPFlag("out", cmd.Flags().Lookup("out")); err != nil {
+				return fmt.Errorf("bind out flag: %w", err)
+			}
+			return nil
+		},
 
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if err := runGenerateCommand(args[0]); err != nil {
+			if err := runGenerateCommand(); err != nil {
 				return fmt.Errorf("generate: %w", err)
 			}
 
 			return nil
 		},
 	}
 
+	cmd.Flags().StringP("out", "o", "alerts.md",
+		"file name or path for the output-file")
+
 	return &cmd
 }
 
-func runGenerateCommand(outputFile string) error {
+func runGenerateCommand() error {
+	out := viper.GetString("out")
+	outputDir, outputFile := path.Split(out)
```

Comment:
We should strive to declare variables as close as possible to the first time that they are being used. In this case, `outputDir` and `outputFile` are declared, followed by `workingDir`.